tests/core/full_node/full_sync/test_full_sync.py

```diff
@@ -412,3 +415,59 @@ async def test_sync_none_wp_response_backward_comp(self, three_nodes, default_10
         assert res is None
         duration = time.time() - start
         assert duration > 5
+
+    @pytest.mark.asyncio
+    async def test_bad_peak_cache(
+        self, three_nodes, default_400_blocks, blockchain_constants, self_hostname, consensus_mode, bt
+    ):
+        if consensus_mode != Mode.PLAIN:
+            pytest.skip("Skipped test")
+        full_node_1, full_node_2, full_node_3 = three_nodes
+        bt.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=1000000)
+        blocks = bt.get_consecutive_blocks(700, default_400_blocks)
+        full_node_2.full_node.blockchain.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=1000000)
+        full_node_1.full_node.blockchain.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=400)
+        for block in blocks:
+            await full_node_2.full_node.add_block(block)
+        server_1 = full_node_1.full_node.server
+        server_2 = full_node_2.full_node.server
+        await server_2.start_client(
+            PeerInfo(self_hostname, uint16(server_1._port)), on_connect=full_node_2.full_node.on_connect
+        )
+        await time_out_assert(60, node_height_between, True, full_node_1, 380, 450)
+        await time_out_assert(60, full_node_1.full_node.sync_store.get_long_sync, False)
+        # call peer has block to populate peer_to_peak
+        peak = full_node_2.full_node.blockchain.get_peak()
+        wp = await full_node_2.full_node.weight_proof_handler.get_proof_of_weight(peak.header_hash)
+        assert full_node_1.full_node.check_bad_peak_cache(wp) is False
+
+    @pytest.mark.asyncio
+    async def test_skip_bad_peak_validation(
```

Comment:
Just a nuanced detail where when I read it as is it sounds like it is going to skip the mechanism that validates bad peaks.

```suggestion
    async def test_skip_validation_of_a_bad_peak(
```

Reply:
i think the name is fine